java/src/test/java/ai/rapids/cudf/ast/CompiledExpressionTest.java

```diff
@@ -0,0 +1,539 @@
+    BinaryExpression expr = new BinaryExpression(op,
+        new ColumnReference(0),
+        new ColumnReference(1));
+    try (Table t = new Table.TestBuilder().column(in1).column(in2).build();
+         CompiledExpression compiledExpr = expr.compile();
+         ColumnVector actual = compiledExpr.computeColumn(t);
+         ColumnVector expected = ColumnVector.fromBoxedBooleans(
+             expectedValues.toArray(new Boolean[0]))) {
+      assertColumnsAreEqual(expected, actual);
+    }
+  }
+
+  private static Stream<Arguments> createBinaryBitwiseExpressionParams() {
+    Integer[] in1 = new Integer[] { -5, 4, null, 2, -3 };
+    Integer[] in2 = new Integer[] { 123, -456, null, 0, -3 };
+    return Stream.of(
+        Arguments.of(AstOperator.BITWISE_AND, in1, in2, mapArray(in1, in2, (a, b) -> a & b)),
+        Arguments.of(AstOperator.BITWISE_OR, in1, in2, mapArray(in1, in2, (a, b) -> a | b)),
+        Arguments.of(AstOperator.BITWISE_XOR, in1, in2, mapArray(in1, in2, (a, b) -> a ^ b)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("createBinaryBitwiseExpressionParams")
+  void testBinaryBitwiseExpressionTransform(AstOperator op, Integer[] in1, Integer[] in2,
+                                            List<Integer> expectedValues) {
+    BinaryExpression expr = new BinaryExpression(op,
+        new ColumnReference(0),
+        new ColumnReference(1));
+    try (Table t = new Table.TestBuilder().column(in1).column(in2).build();
+         CompiledExpression compiledExpr = expr.compile();
+         ColumnVector actual = compiledExpr.computeColumn(t);
+         ColumnVector expected = ColumnVector.fromBoxedInts(
+             expectedValues.toArray(new Integer[0]))) {
+      assertColumnsAreEqual(expected, actual);
+    }
+  }
+
+  private static Stream<Arguments> createBinaryBooleanExpressionParams() {
+    Boolean[] in1 = new Boolean[] { false, true, null, true, false };
+    Boolean[] in2 = new Boolean[] { true, null, null, true, false };
+    return Stream.of(
+        Arguments.of(AstOperator.LOGICAL_AND, in1, in2, mapArray(in1, in2, (a, b) -> a && b)),
+        Arguments.of(AstOperator.LOGICAL_OR, in1, in2, mapArray(in1, in2, (a, b) -> a || b)));
+  }
+
+  @ParameterizedTest
+  @MethodSource("createBinaryBooleanExpressionParams")
+  void testBinaryBooleanExpressionTransform(AstOperator op, Boolean[] in1, Boolean[] in2,
+                                            List<Boolean> expectedValues) {
+    BinaryExpression expr = new BinaryExpression(op,
+        new ColumnReference(0),
+        new ColumnReference(1));
+    try (Table t = new Table.TestBuilder().column(in1).column(in2).build();
+         CompiledExpression compiledExpr = expr.compile();
+         ColumnVector actual = compiledExpr.computeColumn(t);
+         ColumnVector expected = ColumnVector.fromBoxedBooleans(
+             expectedValues.toArray(new Boolean[0]))) {
+      assertColumnsAreEqual(expected, actual);
+    }
+  }
```

Comment:
What happens if we mix types?  Are there operators/ways to do a cast? I didn't see anything like that being tested here.

Reply:
> What happens if we mix types? 

No idea, I was more focused on testing the basic JNI functionality, i.e.: serialization/deserialization, touching the operators and at least one test with each type.

> Are there operators/ways to do a cast?

There are no casting operators yet.  There is some implicit casting done via some of the operators like `TRUE_DIV` and `FLOOR_DIV` which upcast to FLOAT64.  I'll add some tests for mixing types.

Reply:
Turns out type mixing is not supported at all.  Types on a binary expression must match or an error is thrown.  I added a test to check this behavior.  I suspect in the future we could see a cast operator, but for now it is the application's responsibility to ensure all input types are "pre cast" before evaluating the expression.